common/ferrostar/src/routing_adapters/osrm/mod.rs

```diff
@@ -81,6 +81,14 @@ impl Route {
             })
             .collect();
 
+        Self::from_osrm_route(route, &waypoints, polyline_precision)
+    }
+
+    pub fn from_osrm_route(
```

Comment:
Maybe this for a name? Both methods are from OSRM routes 😂 I think you should probably also add a docstring to both now.

```suggestion
    pub fn from_osrm_with_standard_waypoints(
```